### PR TITLE
chore(confirm): convert enzyme tests to RTL

### PR DESCRIPTION
**Pull request by nuria1110**
### Proposed behaviour

Convert `Confirm` tests to RTL.

### Current behaviour

`Confirm` tests are in enzyme. 

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/confirm/confirm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import Confirm from "../confirm";
+
+jest.mock("../../__internal__/utils/helpers/guid", () => () => "guid-12345");
+
+describe("Confirm", () => {
```
Comment (nineteen88) [inline src/components/confirm/confirm.test.tsx:8]: suggestion: remove unnecessary describe block

File: src/components/confirm/confirm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  });
+
+  it("should call onCancel when cancel button is clicked", async () => {
+    const onCancel = jest.fn();
+    const user = userEvent.setup();
+    render(<Confirm open onConfirm={() => {}} onCancel={onCancel} />);
+
+    await user.click(screen.getByRole("button", { name: "No" }));
+    expect(onCancel).toHaveBeenCalledTimes(1);
+  });
```
Comment (nineteen88) [inline src/components/confirm/confirm.test.tsx:46]: suggestion(non-blocking): these four tests should be able to be condensed into two tests

Believe that should be possible anyway, if we lose coverage then obviously not. 

We could also add a more descriptive test title for them as well. e.g.

`it("should render a confirm button when onConfirm prop is set and will execute the function when clicked")`

File: src/components/confirm/confirm.test.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+          "data-element": "bar",
+        }}
+        closeButtonDataProps={{
+          "data-role": "close-button",
+          "data-element": "baz",
+        }}
+      />
+    );
+
+    expect(screen.queryByTestId("confirm-button")).toHaveAttribute(
```
Comment (edleeks87) [inline src/components/confirm/confirm.test.tsx:175]:
```suggestion
    expect(screen.getByRole("button", {name: "Yes")).toHaveAttribute(
    expect(screen.getByRole("button", {name: "No")).toHaveAttribute(

```
(Non-blocking) I think the recommended approach is to use getByRole wherever possible

File: src/components/confirm/confirm.test.tsx (hunk 1 above)
Comment (nineteen88) [reply src/components/confirm/confirm.test.tsx:8]: Should also add that if tests aren't within a describe block then they should be `test()` as opposed to `it()` based on Kent C Dodd's recommendations

See: https://kentcdodds.com/blog/avoid-nesting-when-youre-testing#inline-it

File: src/components/confirm/confirm.test.tsx (hunk 3 above)
Comment (nuria1110) [reply src/components/confirm/confirm.test.tsx:175]: I went with this approach to check that the button was accessible by the new "data-role" being set, but I can change this to use `getByRole` and then assert the data-role attribute is set also correctly if thats preferred.

**Review by edleeks87 — APPROVED** [review]

**Review by nineteen88 — APPROVED** [review]

**Pull request merged**